webapp/tests/Unit/Controller/Jury/JuryControllerTest.php

```diff
@@ -249,6 +252,71 @@ public function testCheckAddEntityAdmin(): void
         }
     }
 
+    /**
+     * Test that admin can add edit an entity for this controller
+     * 
+     * @dataProvider provideEditEntities
+     */
+    public function testCheckEditEntityAdmin(string $identifier, array $formDataKeys, array $formDataValues): void
+    {
+        $this->roles = ['admin'];
+        $this->logOut();
+        $this->logIn();
+        $this->loadFixtures(static::$deleteFixtures);
+        $this->verifyPageResponse('GET', static::$baseUrl, 200);
+        if (static::$edit !== '') {
+            $this->client->followRedirects(true);
+            $crawler = $this->getCurrentCrawler();
+            foreach ($crawler->filter('a') as $node) {
+                if (strpos($node->nodeValue, $identifier) !== false) {
+                    $singlePageLink = $node->getAttribute('href');
+                }
+            }
+            $this->verifyPageResponse('GET', $singlePageLink, 200);
+            $crawler = $this->getCurrentCrawler();
+            foreach ($crawler->filter('a') as $node) {
+                 if (strpos($node->nodeValue, 'Edit') !== false) {
+                    $editLink = $node->getAttribute('href');
+                }
+            }
+            $this->verifyPageResponse('GET', $editLink, 200);
+            $crawler = $this->getCurrentCrawler();
+            foreach ($formDataKeys as $id => $key) {
+                $formFields[static::$addForm . $key . "]"] = $formDataValues[$id];
+            }
+            $button = $this->client->getCrawler()->selectButton('Save');
+            $form = $button->form($formFields, 'POST');
+            $this->client->submit($form);
```

Comment:
Would we need to check that this actually went ok and is not eg an error 500 page?

Reply:
I can implement that,

I think that there is not a case where your POST action would give 500 but the resulting page would have the expected information but thats a bit dependant on where you fail.